swap/src/protocol/alice/event_loop.rs

```diff
@@ -164,8 +169,8 @@ where
 
                             let state0 = match State0::new(btc, xmr, self.env_config, self.bitcoin_wallet.as_ref(), &mut OsRng).await {
                                 Ok(state) => state,
-                                Err(e) => {
-                                    tracing::warn!(%peer, "Failed to make State0 for execution setup: {:#}", e);
+                                Err(error) => {
+                                    tracing::warn!(%peer, %error, "Failed to make State0 for execution setup.");
```

Comment:
This change is not quite functionally equivalent. `{:#}` will log the whole causality chain for an `anyhow::Error` whereas `%` will just invoke regular `Display`. See this open ticket here: https://github.com/tokio-rs/tracing/issues/1311

It is not yet resolved and I think we should really keep the causality chain otherwise debugging might be painful.

Reply:
> It is not yet resolved and I think we should really keep the causality chain otherwise debugging might be painful.

Good catch, I did not know that.

Reply:
I revert and print the error message inside the `message` and unified _somehow_ across all log messages.